core/src/test/java/org/apache/iceberg/TestSnapshotManager.java

```diff
@@ -115,11 +116,10 @@ public void testCherryPickDynamicOverwriteConflict() {
     long lastSnapshotId = table.currentSnapshot().snapshotId();
 
     // pick the snapshot into the current state
-    AssertHelpers.assertThrows(
-        "Should reject partition replacement when a partition has been modified",
```

Comment:
@nastra, why was this merged when it drops the context from all of these assertions? We took the time to write these context strings that explain what the assertion is testing and I would expect them to be carried forward.

Reply:
@rdblue when I reviewed all of these cases the underlying error message was providing enough context to a reader of the test, thus IMHO the additional context felt redundant. Do you see a few places where you'd like to add the additional context back? 
Maybe we should improve the error message in those places where we think some additional context is needed to explain the scenario?